### PR TITLE
8.0 database cleanup backport 9 improvements

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

File: database_cleanup/identifier_adapter.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from psycopg2.extensions import ISQLQuote
+
+
+class IdentifierAdapter(ISQLQuote):
```
Comment (StefanRijnhart) [inline database_cleanup/identifier_adapter.py:7]: How about adding this to openupgradelib and add the dependency in this module?

Comment (StefanRijnhart) [inline database_cleanup/identifier_adapter.py:7]: BTW. just read about AsIs. How does this compare to

```
from psycopg2.extensions import AsIs
self.env.cr.execute('select * from %s', (AsIs('res_partner'),))
```

Comment (hbrunn) [inline database_cleanup/identifier_adapter.py:7]: that's the same as string formatting for the purposes of this module

Comment (StefanRijnhart) [inline database_cleanup/identifier_adapter.py:7]: OK. You gotta explain that to me over a beer next time. For now :+1: ~~when Travis becomes green~~

**Comment by hbrunn** [comment]
travis error unrelated

**Comment by sebalix** [comment]
:+1:

**Comment by NL66278** [comment]
:+1:

**Pull request merged**